cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/distributed/DistributedProgramRuntimeService.java

```diff
@@ -151,7 +151,7 @@ public synchronized RuntimeInfo lookup(ProgramId programId, final RunId runId) {
         continue;
       }
 
-      if (!id.toId().equals(programId)) {
+      if (!id.equals(programId)) {
```

Comment:
You can ask Intellij to find more instances of this warning. I think it'd be worth it to run that check.